libs/design/src/atoms/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit, Input, HostBinding, ChangeDetectionStrategy, forwardRef, EventEmitter, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+let checkboxIdNum = 0;
+
+@Component({
+  selector: 'daff-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DaffCheckboxComponent),
+      multi: true
+    }
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DaffCheckboxComponent {
+
+  /**
+   * The name of the checkbox.
+   */
+  @Input() name: string;
+  /**
+   * The value of the checkbox.
+   */
+  @Input() value: any;
+  /**
+   * Boolean value to determine whether or not the checkbox is checked.
+   */
+  @Input() checked = false;
+  /**
+   * The id of the checkbox. Must be unique. If not entered by a user then it is generated.
+   */
+  @Input() id: string = 'daff-checkbox-' + checkboxIdNum;
+  /**
+   * The aria-label of the checkbox. If not set by user then it defaults to the name of the checkbox.
+   */
+  @Input() label = name;
```

Comment:
Can't we just re-purpose `aria-label` as a pass-through onto the child element? Same for `labeled-by`?